x-pack/plugins/ml/index.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resolve } from 'path';
+import { i18n } from '@kbn/i18n';
+import KbnServer, { Server } from 'src/legacy/server/kbn_server';
+import { plugin } from './server/new_platform';
+import {
+  MlInitializerContext,
+  MlCoreSetup,
+  MlHttpServiceSetup,
+} from './server/new_platform/plugin';
+// @ts-ignore: could not find declaration file for module
+import mappings from './mappings';
+
+interface MlServer extends Server {
+  addAppLinksToSampleDataset: () => {};
+}
+
+export const ml = (kibana: any) => {
+  return new kibana.Plugin({
+    require: ['kibana', 'elasticsearch', 'xpack_main'],
+    id: 'ml',
+    configPrefix: 'xpack.ml',
+    publicDir: resolve(__dirname, 'public'),
+
+    uiExports: {
+      app: {
+        title: i18n.translate('xpack.ml.mlNavTitle', {
+          defaultMessage: 'Machine Learning',
+        }),
+        description: i18n.translate('xpack.ml.mlNavDescription', {
+          defaultMessage: 'Machine Learning for the Elastic Stack',
+        }),
+        icon: 'plugins/ml/ml.svg',
+        euiIconType: 'machineLearningApp',
+        main: 'plugins/ml/app',
+      },
+      styleSheetPaths: resolve(__dirname, 'public/index.scss'),
+      hacks: ['plugins/ml/hacks/toggle_app_link_in_nav'],
+      savedObjectSchemas: {
+        'ml-telemetry': {
+          isNamespaceAgnostic: true,
+        },
+      },
+      mappings,
+      home: ['plugins/ml/register_feature'],
+      injectDefaultVars(server: any) {
```

Comment:
could this be a `Server` or `MlServer`?
If not, does kibana have any types for this? also the `kibana` parameter in the outer function.